scripts/omDomainJSON.py

```diff
@@ -0,0 +1,80 @@
+#
+# Copyright 2023 The Superpower Institute Ltd.
+#
+# This file is part of OpenMethane.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+"""
+Generate a JSON file describing the domain and grid.
+"""
+
+import numpy as np
+import netCDF4 as nc
+from openmethane_prior.omInputs import domainXr, domainProj
+from openmethane_prior.omOutputs import domainJSONOutputPath
+import json
+
+def make_point(x, y):
+    return ([float(x), float(y)])
+
+def write_domain_json(output_file):
+    # Load raster land-use data
+    print("converting domain grid details to JSON")
+
+    domain = {
```

Comment:
Are you happy that we rely on an implicit schema for now? I am ok with it for now, but if we find ourselves transferring lots of different JSON blobs between Python and TS we need a common set of schema. The python world can assert that the right data is written and TS types can be auto-generated.

A problem for future us but just flagging as you likely have more experience with this.

Reply:
Defining some schemas is probably a really good idea. I imagine I'll want to do the same thing (convert netcdf to json) for pretty much every output of the model.

If the netcdf outputs followed a more descriptive / well documented schema, I'd be happy to pretty much transcribe them 1:1.

Alternatively we could use something like JSON Schema to describe the various shapes of data, and I imagine there might even be codegen for Python and TS.

For the moment, I'm happy with an implicit schema because the shape of the data is still in flux and I just need **something** to move forward with the UI work. Once we've seen more of the data we'll be working with maybe we can work on more formal schemas and I'd be happy to come back and refactor this.